### PR TITLE
[needs #424]Deposit Form: Connect modal to header

### PR DESCRIPTION
**Pull request by NRodriguezcuellar**
closes https://github.com/inveniosoftware/invenio-app-rdm/issues/1292
needs  https://github.com/inveniosoftware/react-invenio-deposit/pull/424

### Description
Adds the community selection modal to the community header in the deposit form 
### Checklist
:rotating_light: Please review the [guidelines for contributing](https://invenio.readthedocs.io/en/latest/community/contributing/contribution-guide.html) to Invenio.

- [x] Did you create different commits for logically separate things? Check [the format](https://invenio.readthedocs.io/en/latest/community/contributing/contribution-guide.html#commit-messages) of the commit(s) message(s).
- [x] [InvenioRDM specific] Did you follow the style guide?  Check [the documentation](https://inveniordm.docs.cern.ch/maintenance/styleguide/).
- [x] [InvenioRDM specific] Did you add translations (i18n) for new labels, backend and frontend? Check [the documentation](https://inveniordm.docs.cern.ch/develop/topics/i18n/).
- [x] [InvenioRDM specific] Did you add accessibility (a11y) tags? Check [the documentation](https://inveniordm.docs.cern.ch/maintenance/accessibility/).
- [x] Did you add documentation? We all love well-documented frameworks, right?
- [ ] Did you include third-party code (a library, package, etc.)? If yes:
    - [ ] Make sure that its open source license is compatible with Invenio (*note:* GPL and AGPL licensed code is not accepted)
    - [ ] Did you respect the license conditions (normally you need to keep the copyright and license header)?

> Green light on all GitHub Actions status checks is required in order to merge your PR. Check your code additions will fail neither code linting checks nor tests and will not decrease test coverage.

#### Copyright

- [x] Identify who is the copyright holder (a legal entity - i.e. person or organization - normally this is your employer) for the produced code.
- [x] Make sure that you have updated the copyright headers for touched files (non-trivial contributions >15 lines).

See [copyright and license policy](https://invenio.readthedocs.io/en/latest/community/contributing/maintainers-guide/copyright-policy.html) for details.

#### Reminder

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

If you ticked all the boxes above, you're good to go! :rocket:

### CODE REVIEW

File: src/lib/components/CommunitySelectionModal/CommunitySelectionRSKApp.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
+// This file is part of React-Invenio-Deposit
```
Comment (jrcastro2) [inline src/lib/components/CommunitySelectionModal/CommunitySelectionRSKApp.js:1]: This file looks like a duplication of [CommunitySelectionSearch](https://github.com/inveniosoftware/react-invenio-deposit/blob/2069bbdf2bc6ddcc65f99619fdfceb39c1c6db33/src/lib/components/CommunitySelectionModal/CommunitySelectionSearch.js). Why is this needed?

Comment (NRodriguezcuellar) [reply src/lib/components/CommunitySelectionModal/CommunitySelectionRSKApp.js]: my bad, it is the same component but i changed the name, it is rebased properly now.

File: src/lib/components/Communities/CommunityHeader.js [hunk 2]
```diff
@@ -83,17 +20,48 @@ export class CommunityHeaderComponent extends React.Component {
   }
 
   render() {
-    const { communityRedux, imagePlaceholderLink } = this.props;
+    const { communityRedux, imagePlaceholderLink, setCommunity } = this.props;
```
Comment (zzacharo) [inline src/lib/components/Communities/CommunityHeader.js:23]: Can you rename `communityRedux` back to `community`?

Comment (NRodriguezcuellar) [reply src/lib/components/Communities/CommunityHeader.js:23]: `community` already exists as a prop, which is passed down by jinja, and `communityRedux` is the redux state of the community

Comment (zzacharo) [reply src/lib/components/Communities/CommunityHeader.js:23]: The `community` prop passed from Jinja would be initialized in redux. I had done that in my [draft pr](https://github.com/inveniosoftware/react-invenio-deposit/pull/419/files#diff-fb22c95a502cae93ba1418946e8c3ecf35676cde6f5fd8c665df3c9ec7853d9eR61) for requesting review. Can you merge the community with the communityRedux property?

**Review by zzacharo — APPROVED** [review]
